packages/astro/src/core/render/README.md

```diff
@@ -0,0 +1,68 @@
+# `core/render`
+
+This directory contains most of Astro's high-level rendering APIs, including `renderPage`, `createRenderContext`, etc.
+
+- For rendering an Astro file, see [src/runtime/server/](../../runtime/server/).
+- For rendering an endpoint, see [src/core/endpoint/](../endpoint/).
+
+## Concepts
+
+The codebase has a few abstractions for rendering:
+
+### `RenderContext`
+
+Each render (page or endpoint) requires a `RenderContext` that contains information of:
+
+- The `Request` object
+- The matched `route`
+- The matched `pathname` (without the `base`)
+- The route `params` and `props`
+- Additional `styles`, `links`, `scripts`
+- And more!
+
+It is agnostic to what's being rendered and should only contain per-request information.
```

Comment:
The second half of this seems important and sets up a pattern echoed in the Environment section, so I’d suggest breaking it onto a new line, and maybe some consistent styling between sections:

```suggestion
The `RenderContext` is agnostic to what's being rendered. 

**Permitted state:** `RenderContext` should only contain per-request information.
```